vnext/Microsoft.ReactNative/Fabric/Composition/SwitchComponentView.cpp

```diff
@@ -301,19 +316,20 @@ void SwitchComponentView::onPointerReleased(
   if (!args.GetCurrentPoint(-1).Properties().IsPrimary()) {
     return;
   }
-
   m_pressed = false;
 }
 
 void SwitchComponentView::onPointerEntered(
     const winrt::Microsoft::ReactNative::Composition::Input::PointerRoutedEventArgs &args) noexcept {
   m_hovered = true;
+  m_pointerExited = false;
```

Comment:
How is this different than !m_hovered?

Reply:
I thought that too but m_hoverd is false even when the pointer hasn't been released (ie the mouse hasn't directly interacted with the Switch or when first loaded). I needed this flag for the edge case of the Switch value being changed with the JS (ie not m_hovered or m_pressed) and to figure out when to decrease the size of the thumb.

Reply:
The only difference between m_pointerExited and !m_hovered is the initial state.  And if you need that difference, then there will almost certainly be cases in the non-initial case which are wrong.

Why would you need to change the size if its changed from JS.  The size shouldn't change if it comes from the JS. (or from keyboard.

Reply:
The size doesn't get changed when the switch value is changed in the JS (you can see that in the change event's example in rn-tester) because the code is checking m_pointerExited before changing the position of the thumb. I'm mainly using m_pointerExited to tell if the position of thumb needs to be animated or not (there's not a state currently keeping track of if Switch's value was changed previously if that makes sense!). 

but I can see how that can be confusing and how m_pointerExited may not be correct after the initial case, let me see if I if I can handle these edge cases on the composition side!